olive/cli/base.py

```diff
@@ -43,15 +43,31 @@ def get_model_name_or_path(model_name_or_path) -> Union[str, Dict[str, str]]:
     pattern = r"^(?P<registry_name>[^:]+):(?P<model_name>[^:]+):(?P<version>[^:]+)$"
     match = re.match(pattern, model_name_or_path)
 
-    if not match:
-        return model_name_or_path
+    if match:
+        return {
+            "type": "azureml_registry_model",
+            "registry_name": match.group("registry_name"),
+            "name": match.group("model_name"),
+            "version": match.group("version"),
+        }
+
+    pattern = r"https://huggingface\.co/([^/]+/[^/]+)(?:/.*)?"
```

Comment:
is this one of the customer scenarios? I have usually only seen people provide the id directly. just wondering

Reply:
Yes this is a case we need to support.